test/shared/MockConnection.cs

```diff
@@ -18,7 +18,10 @@ public MockConnection(KestrelServerOptions options)
         {
             ConnectionControl = this;
             RequestAbortedSource = new CancellationTokenSource();
-            ListenerContext = new ListenerContext(new ServiceContext { ServerOptions = options });
```

Comment:
Nit: Classes moved to the shared test folder should be in the `Microsoft.AspNetCore.Testing` namespace.